tests/snuba/api/endpoints/test_organization_events_stats_mep.py

```diff
@@ -1150,3 +1149,67 @@ def test_top_events_with_transaction_on_demand_and_no_environment(self):
             group, agg, row1, row2 = group_count
             row_data = response.data[group][agg]["data"][:2]
             assert [attrs for time, attrs in row_data] == [[{"count": row1}], [{"count": row2}]]
+
+    def test_timeseries_on_demand_with_multiple_percentiles(self):
+        field = "p75(measurements.fcp)"
+        field_two = "p75(measurements.lcp)"
+        groupbys = []
+        query = "transaction.duration:>=100"
+        spec = OnDemandMetricSpec(
+            field=field, groupbys=groupbys, query=query, spec_type=MetricSpecType.DYNAMIC_QUERY
+        )
+        spec_two = OnDemandMetricSpec(
+            field=field_two, groupbys=groupbys, query=query, spec_type=MetricSpecType.DYNAMIC_QUERY
+        )
+
+        assert (
+            spec._query_str_for_hash
+            == "event.measurements.fcp.value;{'name': 'event.duration', 'op': 'gte', 'value': 100.0}"
+        )
+        assert (
+            spec_two._query_str_for_hash
+            == "event.measurements.lcp.value;{'name': 'event.duration', 'op': 'gte', 'value': 100.0}"
+        )
+
+        # Include transaction duration in indexer since it's in the query.
+        indexer.record(UseCaseID.TRANSACTIONS, self.organization.id, "transaction.duration")
+
+        for count in range(0, 4):
+            self.store_on_demand_metric(
+                count * 100,
+                spec=spec,
+                timestamp=self.day_ago + timedelta(hours=1),
+            )
+            self.store_on_demand_metric(
+                count * 200.0,
```

Comment:
Same as above, holding `hours=1`, these will store [0,200,400,..] in the single hour time bucket, which p75 will approximate.